src/test/kotlin/test/de/itemis/mps/gradle/RunMigrationsTest.kt

```diff
@@ -0,0 +1,185 @@
+    private lateinit var settingsFile: File
+    private lateinit var buildFile: File
+    private lateinit var cp: List<File>
+    private lateinit var mpsTestPrjLocation: File
+    
+    private fun commonGradleScriptPart():String {
+        return """
+            import java.net.URI
+            buildscript {
+                dependencies {
+                    "classpath"(files(${cp.map { """"${it.invariantSeparatorsPath}"""" }.joinToString()}))
+                }
+            }
+            
+            plugins {
+                id("run-migrations")
+            }
+            
+            repositories {
+                mavenCentral()
+                maven {
+                    url = URI("https://artifacts.itemis.cloud/repository/maven-mps")
+                }
+            }
+            
+             val mps = configurations.create("mps")
+        """.trimIndent()
+    }
+
+    @Before
+    fun setup() {
+        settingsFile = testProjectDir.newFile("settings.gradle.kts")
+        settingsFile.writeText(
+            """
+            rootProject.name = "hello-world"
+        """.trimIndent()
+        )
+        buildFile = testProjectDir.newFile("build.gradle.kts")
+        cp = javaClass.classLoader.getResource(
+            "plugin-classpath.txt"
+        )!!.readText().lines().map { File(it) }
+        mpsTestPrjLocation = testProjectDir.newFolder("mps-prj")
+        ProjectHelper().extractTestProject(mpsTestPrjLocation)
+    }
+    
+    @Test
+    fun `check run migrations works with MPS 2020_3`() {
+        buildFile.writeText(
+            """
+            ${commonGradleScriptPart()}
+            
+            dependencies {
+                mps("com.jetbrains:mps:2020.3.4")
+            }
+            
+            runMigrations {
+                projectLocation.set(file("${mpsTestPrjLocation.toPath()}"))
+                mpsConfig.set(mps)
+            }
+        """.trimIndent()
+        )
+
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir.root)
+            .withArguments("runMigrations")
+            .withPluginClasspath(cp)
+            .build()
+        Assert.assertEquals(TaskOutcome.SUCCESS, result.task(":runMigrations")?.outcome)
+    }
+
+    @Test
+    fun `check run migrations fails with MPS 2020_3 with force set to true`() {
+        buildFile.writeText(
+            """
+            ${commonGradleScriptPart()}
+            
+            dependencies {
+                mps("com.jetbrains:mps:2020.3.4")
+            }
+            
+            runMigrations {
+                projectLocation.set(file("${mpsTestPrjLocation.toPath()}"))
+                mpsConfig.set(mps)
+                force.set(true)
+            }
+        """.trimIndent()
+        )
+
+        GradleRunner.create()
+            .withProjectDir(testProjectDir.root)
+            .withArguments("runMigrations")
+            .withPluginClasspath(cp)
+            .buildAndFail()
```

Comment:
The failure tests should check that the build failed in the expected place (some such tests passed because the build failed with an unrelated error).